source/codegen/common_helpers.py

```diff
@@ -915,12 +915,25 @@ def get_attribute_groups(data):
     # If the attributes are already in string categories: those are the groups. Return as-is.
     first_key = next(iter(attributes), None)
     if isinstance(first_key, str):
-        return [AttributeGroup(name, attributes, config) for name, attributes in attributes.items()]
+        attribute_groups = [AttributeGroup(name, attributes, config) for name, attributes in attributes.items()]
+        return _transform_attributes(attribute_groups)
 
     # If there's just one level of attributes: use the service_class_prefix as the group.
     service_class_prefix = config["service_class_prefix"]
     return [AttributeGroup(service_class_prefix, attributes, config)]
 
+def _transform_attributes(attribute_groups):
+    # This is handles the specific attributes that uses bitfield enum.
+    # Currently this affects only NI-DAQmx generation.
+    for attribute_group in attribute_groups:
+        for id,data in attribute_group.attributes.items():
+            if "bitfield_enum" in data:
+                data["type"] = BITFIELD_ENUM_RETURN_TYPES.get(data["type"], data["type"])
+    return attribute_groups
```

Comment:
Side effects or return value: choose one.

When I see code like `return _transform_attributes(attribute_groups)`, I think "that must return a transformed copy of the attribute groups". However, `_transform_attributes` modifies the attribute groups that you pass in.

Do you need to mutate the metadata? You can achieve the same result with a non-destructive change in `get_attributes_split_by_sub_group`:

![image](https://github.com/ni/grpc-device/assets/17914399/2bc0cb0a-f9fe-4c53-9795-48e12707bdf9)